### PR TITLE
Add support for Monolog 2

### PR DESCRIPTION
**Pull request by TheoKouzelis**
Closes issue #97 and adds support for Monolog 2.x which is required in Laravel 7.x.

I also have deleted the composer.lock file, to allow composer more freedom to juggle dependencies when people install this package in their projects.

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -1,1929 +0,0 @@
-{
```
Comment (vmihailenco) [inline composer.lock:1]: Any reason to remove this file? It is supposed to be commited with composer.json

Comment (TheoKouzelis) [reply composer.lock:1]: Hi @vmihailenco 

My main description in the PR is wrong. I believed that lock files in libraries would effect projects that depend on this library. And having a lock file specifying Monolog 1.x would stop projects from requiring Monolog 2.x.  But reading the composer docs I've found this not to be true. https://getcomposer.org/doc/02-libraries.md#lock-file

> For your library you may commit the composer.lock file if you want to. This can help your team to always test against the same dependency versions. However, this lock file will not have any effect on other projects that depend on it. It only has an effect on the main project.

However libraries not committing the lock file does seem to be the prevailing convention amongst some of the big libraries in the PHP community. But I suspect the reason for this is down to how they test their library locally and in CI builds. 

* https://github.com/laravel/framework  
* https://github.com/Seldaek/monolog

So in short since this only effects the internals of the library I' happy to go with what you want.

* Should I undo the deletion of the lock file?
* And if so, should I update the lock file to require Monolog 2.x

Comment (vmihailenco) [reply composer.lock:1]: I think this is fine as is - thanks for the contribution!
